zun/container/docker/driver.py

```diff
@@ -343,6 +343,10 @@ def create(self, context, container, image, requested_networks,
                     # TODO: actually look up cgroups for access info
                     host_config['devices'].append(
                         f'{dev["path"]}:{dev["path"]}:rw')
+                host_config["cap_add"] = []
+                for cap_list in oci_config.get("capabilities", {}).values():
```

Comment:
Could be a one-liner using `itertools.chain`, which takes a list of lists and returns one list consisting of a "chain" of all lists you gave it, e.g.: `chain(['A', 'B', 'C'], ['D, 'E', 'F']) => ['A', 'B', 'C', 'D', 'E', 'F']`:

```python
# Using 'set' to trim duplicates
host_config["cap_add"] = set(itertools.chain(oci_config.get("capabilities", {}).values())
```

More readable (no more one-liner ;))

```python
capabilities = oci_config.get("capabilities", {})
all_caps = itertools.chain(capabilities.values())
cap_add = list(set(all_caps))
```